### DISCUSSION TITLE
Sqlalchemy Core select in v1.3

### DISCUSSION THREAD

**Discussion by yawhide** [sqlalchemy/sqlalchemy#7093] [Usage Questions] [answered]
Hello again.

I have a particular situation and I don't understand why it is happening.

I use the `testing.postgresql` library to create a temporary postgresql database. It is pretty straight forward https://github.com/tk0miya/testing.postgresql. It does _not_ run any alembic migrations on its own.

I run my test cases like normal but I get an error from psychopg2 saying `sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "raw_entries" does not exist`. 

This is the weird part... I only get this error if I use the `select` method from sqlalchemy core. If i use sqlalchemy query, i dont get that error. Related code with the model I am using and select query: https://github.com/sqlalchemy/sqlalchemy/discussions/7088

summary:
```
> stmt = select(RawEntrySqlalchemyModel).get('a')
> db.session.execute(stmt)

**boom**

> RawEntrySqlalchemyModel.query.get('a')
> (success)
```

Why does this happen? It seems like if I dont use select, sqlalchemy somehow automatically creates tables in my temporary database. If i use select, it doesnt?

any idea what is going on? I can give more info if you need it.

Thanks again

**Comment by zzzeek** [comment, marked as answer]
well "stmt = select(RawEntrySqlalchemyModel).get('a')" is not SQLAlchemy API, I don't know what that is.

Also "session.execute(stmt)", 1.3 has a session.execute() method but it does *way* less, 1.4 is where that's a way you would be doing ORM queries.    but, i wouldn't think you'd get no such table error.  you'd want to make sure db.session, whatever that is, is pointing to the same database as the one you've created, and that it has started in a new transaction.  if it was already in a transaciton before you created the tables it might not see them (though it should...)
